### ISSUE TITLE
12$: Improve reporting of wrongly ordered or duplicated class member modifiers

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#747]
```js
class C {
  async static f() {}  // error
  static public g() {}  // no error, but should
  async async async async h() {}  // no error, but should
  readonly static x;  // (TypeScript) error
}
```
`parse_and_visit_class_or_interface_member` now has `bump_vector<modifier, ...>` which we can easily analyze to detect misordered and duplicate modifiers. Our current analysis is incomplete.

**Comment by strager** [comment]
We should ideally merge [E0269](https://quick-lint-js.com/errors/E0269/) and [E0232](https://quick-lint-js.com/errors/E0232/).

**Comment by strager** [comment]
`abstract` is a modifier we don't have any ordering diagnostic for.
